Add included to custom rules

bootstraponline opened the pull request:
Fix #689

norio-nomura commented:
Thanks for contribution.
It seems this is work in progress, because `RegexConfiguration.applyConfiguration(_:)` lacks codes for supporting `included`.

norio-nomura commented on Source/SwiftLintFramework/Rules/CustomRules.swift:
```diff
@@ -50,11 +50,25 @@ public struct CustomRules: Rule, ConfigurationProviderRule {
+                let included = $0.included.pattern
+                if included.isEmpty { return true }
+
+                let pathMatch = file.path?.rangeOfString(included,
```
`String.rangeOfString(_:options:range:locale:)` creates `NSRegularExpression` on every call.
We can reuse `$0.included` instead of that.

bootstraponline replied:
Fixed.

bootstraponline commented:
Thanks for the review. I've updated the pull request.

bootstraponline commented:
Any other changes/feedback?

jpsim commented:
Could you please add documentation for this in the readme?

jpsim commented on CHANGELOG.md:
```diff
@@ -6,7 +6,9 @@
 ##### Enhancements
 
-* None.
+* Add included regex for custom rules to control what files are processed.
```
missing two trailing spaces according to our syntax outlined in CONTRIBUTING.md

jpsim commented:
Thanks for doing this @bootstraponline! I didn't do a full review, but quickly glancing through everything, this looks great!

bootstraponline commented:
> Could you please add documentation for this in the readme?

Updated.

> missing two trailing spaces according to our syntax outlined in CONTRIBUTING.md

Fixed.

jpsim commented on Source/SwiftLintFramework/Rules/RuleConfigurations/RegexConfiguration.swift:
```diff
@@ -43,6 +44,10 @@ public struct RegexConfiguration: RuleConfiguration, Equatable {
         regex = try NSRegularExpression.cached(pattern: regexString)
 
+        if let includedString = configurationDict["included"] as? String {
+          included = try NSRegularExpression.cached(pattern: includedString)
```
we use 4-space indentation everywhere else in this project. Could you do that here too?

bootstraponline replied:
Fixed.

jpsim commented on Source/SwiftLintFramework/Rules/CustomRules.swift:
```diff
@@ -50,11 +50,25 @@ public struct CustomRules: Rule, ConfigurationProviderRule {
 
-        return configuration.customRuleConfigurations.flatMap {
+        if let path = file.path {
+            configurations = configurations.filter {
```
could you please use a named block parameter here instead of `$0`? Maybe `{ configuration in`

bootstraponline replied:
Fixed.

jpsim commented:
I think this is good to go once you've addressed my last 2 comments. Thanks!

jpsim commented:
Thanks for your contribution @bootstraponline!

norio-nomura commented:
🎉
